pkg/apis/mysql/v1alpha1/types.go

```diff
@@ -0,0 +1,419 @@
+	// group replication.
+	MaxInnoDBClusterMembers = 9
+
+	// ClusterNameMaxLen is the maximum supported length of a
+	// Cluster name.
+	// See: https://bugs.mysql.com/bug.php?id=90601
+	ClusterNameMaxLen = 28
+)
+
+// TODO (owain) we need to remove this because it's not reasonable for us to maintain a list
+// of all the potential MySQL versions that can be used and in reality, it shouldn't matter
+// too much. The burden of this is not worth the benfit to a user
+var validVersions = []string{
+	defaultVersion,
+}
+
+// ClusterSpec defines the attributes a user can specify when creating a cluster
+type ClusterSpec struct {
+	// Version defines the MySQL Docker image version.
+	Version string `json:"version"`
+
+	// Replicas defines the number of running MySQL instances in a cluster
+	Replicas int32 `json:"replicas,omitempty"`
+
+	// BaseServerID defines the base number used to create uniq server_id for MySQL instances in a cluster.
+	// The baseServerId value need to be in range from 1 to 4294967286
+	// If ommited in the manifest file, or set to 0, defaultBaseServerID value will be used.
+	BaseServerID uint32 `json:"baseServerId,omitempty"`
+
+	// MultiMaster defines the mode of the MySQL cluster. If set to true,
+	// all instances will be R/W. If false (the default), only a single instance
+	// will be R/W and the rest will be R/O.
+	MultiMaster bool `json:"multiMaster,omitempty"`
+
+	// NodeSelector is a selector which must be true for the pod to fit on a node.
+	// Selector which must match a node's labels for the pod to be scheduled on that node.
+	// More info: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/
+	// +optional
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+
+	// If specified, affinity will define the pod's scheduling constraints
+	// +optional
+	Affinity *corev1.Affinity `json:"affinity,omitempty"`
+
+	// VolumeClaimTemplate allows a user to specify how volumes inside a MySQL cluster
+	// +optional
+	VolumeClaimTemplate *corev1.PersistentVolumeClaim `json:"volumeClaimTemplate,omitempty"`
+
+	// BackupVolumeClaimTemplate allows a user to specify a volume to temporarily store the
+	// data for a backup prior to it being shipped to object storage.
+	// +optional
+	BackupVolumeClaimTemplate *corev1.PersistentVolumeClaim `json:"backupVolumeClaimTemplate,omitempty"`
+
+	// If defined, we use this secret for configuring the MYSQL_ROOT_PASSWORD
+	// If it is not set we generate a secret dynamically
+	// +optional
+	SecretRef *corev1.LocalObjectReference `json:"secretRef,omitempty"`
+
+	// ConfigRef allows a user to specify a custom configuration file for MySQL.
+	// +optional
+	ConfigRef *corev1.LocalObjectReference `json:"configRef,omitempty"`
+
+	// SSLSecretRef allows a user to specify custom CA certificate, server certificate
+	// and server key for group replication SSL
+	// +optional
+	SSLSecretRef *corev1.LocalObjectReference `json:"sslSecretRef,omitempty"`
```

Comment:
`s/SSLSecretRef/RootPasswordSecret/g`

Reply:
s/SSLSecretRef/SSLSecret/g but i'm guessing that's what you meant ;)